backend/routes/commentRoutes.js

```diff
@@ -1,11 +1,11 @@
 import express from "express";
 import * as commentController from "../controllers/commentController.js";
 import { authenticateUser } from "../middleware/auth.js";
-
+import { authenticateProfilePermissions } from "../middleware/profile.js";
 const router = express.Router();
 
 // All comment routes require authentication
-router.get("/comments/:profileId", authenticateUser, commentController.queryComments);
+router.get("/comments/:userId", authenticateUser, authenticateProfilePermissions, commentController.queryComments);
 router.post("/comments", authenticateUser, commentController.addComment);
 router.put("/comment/:id", authenticateUser, commentController.editComment);
```

Comment:
I would think you'd want some more auth checks on updating/deleting a comment - maybe a user can only update/delete their own comments, or only admins can update/delete any comment?